src/compiler/checker.ts

```diff
@@ -4873,7 +4907,8 @@ namespace ts {
                     // on either side that need to be prioritized. For example, A | B = (A | B) & (C | D) or
                     // A & B = (A & B) | (C & D).
                     if (source.flags & TypeFlags.Intersection) {
-                        // If target is a union type the following check will report errors so we suppress them here
+                        // If target is a union type then the check following this one will report errors,

```

Comment:
It does if you have more context. "The following check" implies "the one directly below"